backend/libs/validate-auth/src/validate-auth.decorator.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ClientProxy,
+  ClientProxyFactory,
+  Transport,
+} from "@nestjs/microservices";
+
+export function ValidateAuth(): MethodDecorator {
+  return function(
+    target: any,
+    propertyKey: string | symbol,
+    descriptor: PropertyDescriptor,
+  ): PropertyDescriptor {
+    console.log("decorator called");
+    // const originalMethod = descriptor.value;
```

Comment:
System architect will decide where the RMQ port will be exposed and developer will implement it. As this is an auth package we will just define the Event Topic for message routing. Consumer and Producer will be configured by the developers/